packages/app/src/app/pages/Dashboard/Content/routes/Settings/UserSettings/WorkspaceSettings.tsx

```diff
@@ -25,6 +27,42 @@ export const WorkspaceSettings = () => {
   const { user, activeTeam, activeTeamInfo } = useAppState();
   const actions = useActions();
 
+  const [editing, setEditing] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const [file, setFile] = useState<{ name: string; url: string } | null>(null);
+
+  const getFile = async avatar => {
+    const url = await new Promise(resolve => {
+      const reader = new FileReader();
+      reader.onload = e => {
+        resolve(e.target.result);
+      };
+      reader.readAsDataURL(avatar);
+    });
+
+    const stringUrl = url as string;
+
+    setFile({
+      name: avatar.name,
+      url: stringUrl,
+    });
+  };
+
+  const onSubmit = async event => {
+    event.preventDefault();
+    setLoading(true);
+    try {
+      await actions.dashboard.updateTeamAvatar({
+        ...file,
+        teamId: activeTeam,
+      });
+      setLoading(false);
+      setEditing(false);
+    } catch {
+      setLoading(false);
+    }
```

Comment:
**nitpick:** Since we're `setLoading(false);` in both cases, the `catch` can be replaced with a `finally` 😄 

```suggestion
      setEditing(false);
    } finally {
      setLoading(false);
    }
```